src/plugins/dashboard/public/application/app.tsx

```diff
@@ -11,30 +11,48 @@
 
 import './app.scss';
 import { AppMountParameters } from 'opensearch-dashboards/public';
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { Route, Switch, useLocation } from 'react-router-dom';
 import { DashboardConstants, createDashboardEditUrl } from '../dashboard_constants';
 import { DashboardEditor, DashboardListing, DashboardNoMatch } from './components';
+import { useOpenSearchDashboards } from '../../../opensearch_dashboards_react/public';
+import { DashboardServices } from '../types';
+import { syncQueryStateWithUrl } from '../../../data/public';
 
 export interface DashboardAppProps {
   onAppLeave: AppMountParameters['onAppLeave'];
 }
 
 export const DashboardApp = ({ onAppLeave }: DashboardAppProps) => {
+  const {
+    services: {
+      data: { query },
+      osdUrlStateStorage,
+    },
+  } = useOpenSearchDashboards<DashboardServices>();
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    // syncs `_g` portion of url with query services
+    const { stop } = syncQueryStateWithUrl(query, osdUrlStateStorage);
+
+    return () => stop();
+
+    // this effect should re-run when pathname is changed to preserve querystring part,
+    // so the global state is always preserved
+  }, [query, osdUrlStateStorage, pathname]);
+
   return (
     <Switch>
-      <Route exact path={['/', DashboardConstants.LANDING_PAGE_PATH]}>
-        <DashboardListing />
-      </Route>
-      <Route
-        exact
-        path={[DashboardConstants.CREATE_NEW_DASHBOARD_URL, createDashboardEditUrl(':id')]}
-      >
+      <Route path={[DashboardConstants.CREATE_NEW_DASHBOARD_URL, createDashboardEditUrl(':id')]}>
```

Comment:
Yea this can prob be polished better. Originally I wish to do it like:
```
<Switch>
   <Route path= ...> 
        <DashboardEditor>
   </Route>

   <Route path= ...> 
        <DashboardListing>
   </Route>

   <Route path= ...> 
        <DashboardNoMatch>
   </Route>

</Switch>
```

However, since we we are rendering the dashboard page by calling the `render()` function of `dashboard container`: `dashboardContainer.render(dashboardDom);` and we need to pass in a HTML element here, so i added `<div id="dashboardViewport" />` so we can refer the <div> element by its id and pass it to the render function for it to be rendered or updated. There might be a better way but will need to look more into it.